src/qutip_qip/decompositions/single_decompositions.py

```diff
@@ -0,0 +1,226 @@
+
+    # find alpha, beta and theta
+    alpha = cmath.phase(-a_star) + cmath.phase(b_star)
+    beta = cmath.phase(-a_star) - cmath.phase(b_star)
+    theta = 2*np.arctan2(np.absolute(b_star), np.absolute(a))
+    return(alpha, theta, beta, np.pi+global_phase_angle)
+
+
+
+def _ZYZ_rotation(input_gate, num_of_qubits, target):
+    r""" An input 1-qubit gate is expressed as a product of rotation matrices
+    :math:`\textrm{R}_z` and :math:`\textrm{R}_y`.
+
+    Parameters
+    ----------
+    input_gate : :class:`qutip.Qobj`
+        The matrix that's supposed to be decomposed should be a Qobj.
+    """
+    angle_list = _angles_for_ZYZ(input_gate, single_qubit)
+    alpha = angle_list[0]
+    beta = angle_list[2]
+    theta = angle_list[1]
+    global_phase_angle=angle_list[3]
+
+    # for string in circuit diagram
+    alpha_string = alpha/np.pi
+    beta_string = beta/np.pi
+    theta_string = theta/np.pi
+    global_phase_angle_string = global_phase_angle/np.pi
+
+    Phase_gate = Gate("GLOBALPHASE",targets=[target], arg_value=global_phase_angle, arg_label=r'{:0.2f} \times \pi'.format(global_phase_angle_string))
+    Rz_alpha = Gate("RZ",targets=[target], arg_value=alpha, arg_label=r'{:0.2f} \times \pi'.format(alpha_string))
+    Ry_theta = Gate("RY",targets=[target], arg_value=theta, arg_label=r'{:0.2f} \times \pi'.format(theta_string))
+    Rz_beta = Gate("RZ",targets=[target], arg_value=beta, arg_label=r'{:0.2f} \times \pi'.format(beta_string))
+
+    return(Rz_alpha, Ry_theta, Rz_beta, Phase_gate)
+
+def _ZXZ_rotation(input_gate, num_of_qubits, target):
+    r""" An input 1-qubit gate is expressed as a product of rotation matrices
+    :math:`\textrm{R}_z` and :math:`\textrm{R}_x`.
+
+    Parameters
+    ----------
+    input_gate : :class:`qutip.Qobj`
+        The matrix that's supposed to be decomposed should be a Qobj.
+    """
+    angle_list = _angles_for_ZYZ(input_gate, single_qubit)
+    alpha = angle_list[0]
+    alpha = alpha - np.pi/2
+    beta = angle_list[2]
+    beta = beta + np.pi/2
+    theta = angle_list[1]
+    global_phase_angle=angle_list[3]
+
+    # for string in circuit diagram
+    alpha_string = alpha/np.pi
+    beta_string = beta/np.pi
+    theta_string = theta/np.pi
+    global_phase_angle_string = global_phase_angle/np.pi
+
+    Phase_gate = Gate("GLOBALPHASE",targets=[target], arg_value=global_phase_angle, arg_label=r'{:0.2f} \times \pi'.format(global_phase_angle_string))
+    Rz_alpha = Gate("RZ",targets=[target], arg_value=alpha, arg_label=r'{:0.2f} \times \pi'.format(alpha_string))
+    Rx_theta = Gate("RX",targets=[target], arg_value=theta, arg_label=r'{:0.2f} \times \pi'.format(theta_string))
+    Rz_beta = Gate("RZ",targets=[target], arg_value=beta, arg_label=r'{:0.2f} \times \pi'.format(beta_string))
+
+    return(Rz_alpha, Rx_theta, Rz_beta, Phase_gate)
+
+
+_rotation_matrices_dictionary ={"ZYZ": _ZYZ_rotation,
+                                "ZXZ": _ZXZ_rotation,
+                                }
```

Comment:
The routine could be like this (just my thought, not that you have to):
- Regardless of the scheme, always use ZYZ decomposition, get the 3 angles and the global phase.
- For each scheme, there is a private function to decide what is the 3 rotation gates to use (RZ, RY or RZ) as well as the global phase correction. They always use the same rotation angles calculated for ZYZ (maybe add a minus sign here and there if needed). This can be manually derived as I sketched above. 

In this way one only need one function to calculate the ZYZ angles, the rest is just tranformation.

Reply:
Yes, that's what I attempted to do by creating [`_angles_for_ZYZ`](https://github.com/qutip/qutip-qip/pull/67/files#diff-6367c4af96a402daf1a20673bc8e9eb34b0390fd18dc119373485b5105f3b9d4R14). 

I had calculated the angles for the new combinations by using output of ZYZ and transforming Rz as product of RxRyRx and so on..  I made a mistake somewhere in these transformations and the outputs don't match. 

Adding the combinations later down the line is just a matter of re-checking these calculations.